include/sparrow/buffer/dynamic_bitset/dynamic_bitset.hpp

```diff
@@ -92,4 +109,81 @@ namespace sparrow
         : base_type(storage_type(p, this->compute_block_count(n)), n, null_count)
     {
     }
-}
+
+
+    using validity_bitmap = dynamic_bitset<std::uint8_t>;
+
+    
+    namespace detail
+    {
+        using validity_bitmap = sparrow::validity_bitmap;
+        inline validity_bitmap ensure_validity_bitmap_impl(std::size_t size, const validity_bitmap & bitmap)
+        {
+            if(bitmap.size() == 0)
+            {
+                return validity_bitmap(size, true);
+            }
+            return bitmap; // copy
+        }
+    
+        inline validity_bitmap ensure_validity_bitmap_impl(std::size_t size, validity_bitmap && bitmap)
+        {
+            if(bitmap.size() == 0)
+            {
+                bitmap.resize(size, true);
+            }
+            return std::move(bitmap);
+        }
+
+        // range of booleans
+        template <std::ranges::input_range R>
+        requires(std::same_as<std::ranges::range_value_t<R>, bool>)
+        validity_bitmap ensure_validity_bitmap_impl(std::size_t size, R&& range)
+        {   
+            SPARROW_ASSERT_TRUE(size == std::ranges::size(range) || std::ranges::size(range) == 0);
+            validity_bitmap bitmap(size, true);
+            std::size_t i = 0;
+            for(auto value : range)
+            {
+                if(!value){
+                    bitmap.set(i, false);
+                }
+                i++;
+            }
+            return bitmap;
+        }
+
+        // range of indices / integers (but not booleans)
+        template <std::ranges::input_range R>
+        requires(
+            std::unsigned_integral<std::ranges::range_value_t<R>> &&
+            !std::same_as<std::ranges::range_value_t<R>, bool> &&
+            !std::same_as<std::decay_t<R>, validity_bitmap> 
+        )
+        validity_bitmap ensure_validity_bitmap_impl(std::size_t size, R&& range_of_indices)
+        { 
+            validity_bitmap bitmap(size, true);
+            for(auto index : range_of_indices)
+            {
+                bitmap.set(index, false);
+            }
+            return bitmap;
+        }
+    } // namespace detail
+
+    // concept to handle what can be passed to ensure_validity_bitmap
+    template <class T>
+    concept validity_bitmap_input = 
+        std::same_as<T, validity_bitmap&&> ||
+        std::same_as<T, const validity_bitmap&> ||
+        (std::ranges::input_range<T> && std::same_as<std::ranges::range_value_t<T>, bool>) ||
+        (std::ranges::input_range<T> && std::unsigned_integral<std::ranges::range_value_t<T>> );
+
+
+    template <validity_bitmap_input R>
+    validity_bitmap ensure_validity_bitmap(std::size_t size, R&& validity_input)
```

Comment:
If you pass an `validity_input` as an rvalue reference, R will be deduced as the "raw type R", not as the rvalue reference.
So the `validity_bitmap_input` concept should rather be defined as:
```cpp
    template <class T>
    concept validity_bitmap_input = 
        std::same_as<T, validity_bitmap> || // Not validity_bitmap&&
        std::same_as<T, const validity_bitmap&> ||
        (std::ranges::input_range<T> && std::same_as<std::ranges::range_value_t<T>, bool>) ||
        (std::ranges::input_range<T> && std::unsigned_integral<std::ranges::range_value_t<T>> );
```